### PR TITLE
SONAR-10080 Add "My Projects" and "My Issues" pages

### PR DESCRIPTION
**Pull request by stas-vilchik**

### CODE REVIEW

File: server/sonar-web/src/main/js/apps/issues/components/App.js [hunk 1]
```diff
@@ -754,22 +726,16 @@ export default class App extends React.PureComponent {
   }
 
   renderFacets() {
-    const { component, currentUser } = this.props;
+    const { component, currentUser, myIssues } = this.props;
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/issues/components/App.js]: MAJOR Remove this useless assignment to local variable "currentUser" [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS1854)
MINOR Remove the declaration of the unused 'currentUser' variable. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AUnusedVariable)

File: server/sonar-web/src/main/js/apps/projects/components/PageSidebar.tsx [hunk 2]
```diff
@@ -39,26 +37,26 @@ import TagsFilter from '../filters/TagsFilter';
...
     .filter(key => !['view', 'visualization', 'sort'].includes(key))
     .some(key => query[key] != null);
   const isLeakView = view === 'leak';
-  const basePathName = organization ? `/organizations/${organization}/projects` : '/projects';
-  const pathname = basePathName + (isFavorite ? '/favorite' : '');
   const maxFacetValue = getMaxFacetValue(facets);
-  const facetProps = { isFavorite, maxFacetValue, organization, query };
+  const facetProps = { onQueryChange, maxFacetValue, organization, query };
 
   let linkQuery: RawQuery | undefined = undefined;
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/projects/components/PageSidebar.tsx]: MAJOR Remove this useless assignment to local variable "linkQuery". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1854)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-web/src/main/js/apps/projects/components/App.tsx [hunk 3]
```diff
@@ -18,58 +18,309 @@
...
   }
-}
 
-const mapStateToProps = (state: any) => ({
-  currentUser: getCurrentUser(state),
-  languages: getLanguages(state),
-  organizationsEnabled: areThereCustomOrganizations(state)
-});
+  updateLocationQuery = (newQuery: RawQuery) => {
+    // omit `null`, `undefined` and empty string
```
[the remark this answers is not in the document]
Comment (stas-vilchik) [reply server/sonar-web/src/main/js/apps/projects/components/App.tsx]: True. Fortunately we don't have `0` value in facets.

File: server/sonar-web/src/main/js/apps/projects/filters/Filter.tsx [hunk 4]
```diff
@@ -122,12 +125,7 @@ export default class Filter extends React.PureComponent<Props> {
       option > value;
 
     return (
-      <Link
-        key={option}
-        className={className}
-        to={path}
-        data-key={option}
-        onClick={this.blurOnClick}>
+      <a className={className} data-key={option} href="#" key={option} onClick={this.handleClick}>
```
[the remark this answers is not in the document]
Comment (stas-vilchik) [reply server/sonar-web/src/main/js/apps/projects/filters/Filter.tsx]: yes, it is used https://github.com/SonarSource/sonarqube/pull/2832/files/f9337ab054aedf35122465a981f8c7e5400dcac9#diff-1b62db87b12111d680207a382677c67bR73

**Pull request merged**